### PR TITLE
refactor(api): extract bearer-auth concern + core ApiError envelope

### PR DESCRIPTION
**Pull request by sh1nj1**
## What

Server-side DRY refactor of the API layer:

1. Extracts the byte-identical bearer-auth code (`extract_bearer_token`, `current_user`) plus a shared non-rendering `find_user_by_bearer_token(token)` into a new `Collavre::Api::Authenticatable` concern, included by both API base controllers (`collavre` core and `collavre_completion_api`).
2. Introduces `Collavre::ApiError` + a `rescue_from` in the **collavre-core** base controller, so auth failures raise instead of scattering `render json:` calls.

## Why

The Doorkeeper lookup sequence was duplicated across two engines. This removes the duplication while keeping each engine's own error-envelope render (they are external contracts and intentionally differ).

## Wire shapes preserved (this is a pure server refactor — no client changes)

Each is pinned by a test:
1. **422 `{errors: [...]}`** validation array — untouched (JS `queue_manager.js` keys its retry decision on this).
2. **OpenAI `{error: {message, type, code}}`** 400/401 in `collavre_completion_api` — preserved exactly.
3. **Voice `{reply, speak, action}`** — `render_speak` untouched.
4. **SSE streaming error path** in `completions_controller.rb` — the new `rescue_from` is `ApiError`-only and in the *core* engine, so it cannot intercept it.

**Zero `.js` files changed.**

## Notes / decisions

- The concern's `find_user_by_bearer_token` returns user-or-nil (no rendering); each base controller owns its render, so the two envelopes stay distinct.
- `ApiError` + `rescue_from` is scoped to the **collavre-core** engine only. The 422 array renders and the just-hardened `agents_controller` claim path are **not** routed through it.
- One edge-case 401 message ("User not found", accessible-token-but-missing-user) collapses into "Invalid authentication token". No test or client branches on that string.
- i18n of API auth copy is a deliberate follow-up (kept as-is to avoid changing wire output).

## Tests

New 401 body-shape assertions in `agents_controller_test.rb`; existing OpenAI/voice/completions tests pass unchanged. Non-system suites green locally; system tests validated by CI.

### CODE REVIEW

File: engines/collavre_completion_api/app/controllers/collavre_completion_api/api/v1/base_controller.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ module CollavreCompletionApi
   module Api
     module V1
       class BaseController < ActionController::API
+        include Collavre::Api::Authenticatable
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre_completion_api/app/controllers/collavre_completion_api/api/v1/base_controller.rb]: Fixed in 69bb6b6d. You're right — under `USE_COLLAVRE_GEM=true` core resolves to the published gem, which may predate `Collavre::Api::Authenticatable`, so `collavre_completion_api` including it would raise NameError. I kept the concern in the core engine (still used by core's own `api/v1/base_controller`) but reverted `completion_api`'s base controller to inline `extract_bearer_token`/`find_user_by_bearer_token`/`current_user`, so the engine stays loadable against a gem-mode core. completion_api + core auth tests pass.
